debezium-reactive/src/main/java/io/debezium/reactive/internal/RowSendingEmbeddedEngine.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.reactive.internal;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import io.debezium.annotation.ThreadSafe;
+import io.debezium.config.Configuration;
+import io.debezium.embedded.StopConnectorException;
+import io.debezium.embedded.internal.AbstractEmbeddedEngine;
+import io.debezium.embedded.spi.OffsetCommitPolicy;
+import io.debezium.util.Clock;
+
+/**
+ * A default implementation of EmbeddedEngine
+ *
+ * @author Randall Hauch, Jiri Pechanec
```

Comment:
was this copied from somewhere  ? if so, perhaps some more abstract/base is needed ?